### PR TITLE
[Test] Validate gluster block provisioner in multiple clusters

### PR DESCRIPTION
**Pull request by srivickynesh**
Check gluster block provisioner creation across multiple cluster

Signed-off-by: Sri Vignesh <sselvan@redhat.com>

### CODE REVIEW

**Review by vamahaja — APPROVED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 1]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
...
+                "Config file doesn't have key"
+                "'storage_classes.registry_block_storage_class.resturl' or "
+                "'storage_classes.registry_block_storage_class"
+                ".restsecretnamespace'")
+
+        size = 1
+        prefix = 'autotest-pvc-{}'.format(utils.get_random_str(size=5))
+
+        # Create PVC and verify multipath in glusterfs namespace
+        switch_oc_project(self.node, 'glusterfs')
```
Comment (SushilG96) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: Add cleanup.
Is it not that glusterfs the default project?

**Review by SushilG96 — CHANGES_REQUESTED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 1 above)
Comment (srivickynesh) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py]: Please refer this: https://review.gluster.org/#/c/glusterfs-containers-tests/+/25221/9/tests/functional/provisioning/test_dynamic_provisioning_block.py@890

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 2]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
             # Delete the PVC
             oc_delete(self.node, 'pvc', pvc_name)
             wait_for_resource_absence(self.node, 'pvc', pvc_name)
+
+    @pytest.mark.tier2
+    def test_block_provisioner_multiple_custers(self):
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: test_block_provisioner_on_multiple_clusters

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 1 above)
Comment (SushilG96) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py:900]: I think its better to remove it, as It's already in the baseclass.

**Review by SushilG96 — CHANGES_REQUESTED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 1 above)
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: Get storage project dynamically.

Comment (SushilG96) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py]: I think it's better to remove it, as we are already switching in baseclass

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 3]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
...
+        switch_oc_project(self.node, 'glusterfs')
+        pvc_name = self.create_and_wait_for_pvc(
+            pvc_size=size, pvc_name_prefix=prefix)
+        dc_name, pod_name = self.create_dc_with_pvc(pvc_name)
+        match_pvc_and_pv(self.node, prefix)
+        self.verify_iscsi_sessions_and_multipath(
+            pvc_name, pod_name, rtype='pod')
+
+        # Change project namespace
+        self.addCleanup(switch_oc_project, self.node, 'glusterfs')
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: Get storage project dynamically.

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 4]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
...
+                ".restsecretnamespace'")
+
+        size = 1
+        prefix = 'autotest-pvc-{}'.format(utils.get_random_str(size=5))
+
+        # Create PVC and verify multipath in glusterfs namespace
+        switch_oc_project(self.node, 'glusterfs')
+        pvc_name = self.create_and_wait_for_pvc(
+            pvc_size=size, pvc_name_prefix=prefix)
+        dc_name, pod_name = self.create_dc_with_pvc(pvc_name)
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: dc_name is not used anywhere.

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 5]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
...
+
+        # Change project namespace
+        self.addCleanup(switch_oc_project, self.node, 'glusterfs')
+        switch_oc_project(self.node, self._registry_project_name)
+
+        # Create PVC and verify multipath in registry namespace
+        self.sc_name = self.create_storage_class(glusterfs_registry=True)
+        pvc_name = self.create_and_wait_for_pvc(
+            sc_name=self.sc_name, pvc_size=size, pvc_name_prefix=prefix)
+        dc_name, pod_name = self.create_dc_with_pvc(pvc_name)
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: dc_name is not used anywhere.

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 1 above)
Comment (vamahaja) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py]: and update comment as `Create PVC in default namespace and verify multipath`
`

**Review by adityaramteke — CHANGES_REQUESTED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 3 above)
Comment (adityaramteke) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py]: This comment is not addressed.

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 6]
```diff
@@ -868,3 +869,51 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
             # Delete the PVC
             oc_delete(self.node, 'pvc', pvc_name)
             wait_for_resource_absence(self.node, 'pvc', pvc_name)
+
+    @pytest.mark.tier2
+    def test_block_provisioner_on_multiple_custers(self):
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py]: Custers or clusters?

**Review by adityaramteke — CHANGES_REQUESTED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 3 above)
Comment (adityaramteke) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py]: Please replace 'glusterfs' with self.storage_project_name

File: tests/functional/provisioning/test_dynamic_provisioning_block.py [hunk 7]
```diff
@@ -868,3 +869,52 @@ def test_100gb_block_pvc_create_and_delete_twice(self):
...
+            pvc_size=size, pvc_name_prefix=prefix)
+        _, pod_name = self.create_dc_with_pvc(pvc_name)
+        match_pvc_and_pv(self.node, prefix)
+        self.verify_iscsi_sessions_and_multipath(
+            pvc_name, pod_name, rtype='pod')
+
+        # Change project namespace
+        self.addCleanup(
+            switch_oc_project, self.node, self.storage_project_name)
+        switch_oc_project(self.node, self._registry_project_name)
```
Comment (adityaramteke) [inline tests/functional/provisioning/test_dynamic_provisioning_block.py:910]: I can't find _registry_project_name in tests/glusterfs-containers-tests-config.yaml, can you please check and correct it. I think it is registry_project_name.

**Review by adityaramteke — CHANGES_REQUESTED** [review]

File: tests/functional/provisioning/test_dynamic_provisioning_block.py (hunk 7 above)
Comment (srivickynesh) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py:910]: See line number 886,
        self._registry_project_name = self.registry_sc.get(
            'restsecretnamespace')

Comment (adityaramteke) [reply tests/functional/provisioning/test_dynamic_provisioning_block.py:910]: LGTM

**Review by adityaramteke — APPROVED** [review]

**Review by SushilG96 — APPROVED** [review]
LGTM

**Pull request merged**